.github/workflows/kiali-ci.yml

```diff
@@ -32,7 +55,101 @@ jobs:
         run: make swagger-ci
 
       - name: Build
-        run: make clean build
+        run: make clean-all build
 
-      - name: Test
+      - name: Test backend
         run: make test-race
+
+      - name: Upload go binary
+        uses: actions/upload-artifact@v3
+        with:
+          name: kiali
+          path: ~/go/bin/kiali
+
+  build_frontend:
+    name: Build and unit test frontend
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup node
+        uses: actions/setup-node@v2
+        with:
+          node-version: "14"
+          cache: yarn
+          cache-dependency-path: frontend/yarn.lock
+
+      - name: Build
+        run: make clean-all build-ui
```

Comment:
It's fine because different jobs have totally independent workspaces. The only way to share artifacts between them is to upload/download the artifacts which is what we do in each build job. Probably don't need to clean here but it's just a precaution.